.pnpmfile.cjs

```diff
@@ -18,6 +18,25 @@ function readPackage(pkg) {
     pkg.peerDependencies["react"] = "*";
   }
 
+  // patch peer dependencies of packages from `itwinjs-core` to make sure that they are resolved to same version.
+  // peer dependencies in those packages are defined with `^` and that allows to resolve some of them to different version.
+  if (corePackages.includes(pkg.name) && pkg.peerDependencies) {
+    const fixedPeers = Object.entries(pkg.peerDependencies).reduce(
+      (deps, [name, version]) => ({ ...deps, [name]: corePackages.includes(name) ? version.replaceAll("^", "") : version }),
+      {},
+    );
+    pkg.peerDependencies = fixedPeers;
+  }
+
+  // patch peer dependencies of packages from `appui` to make sure that they are resolved to same version.
+  if (appuiPackages.includes(pkg.name) && pkg.peerDependencies) {
+    const fixedPeers = Object.entries(pkg.peerDependencies).reduce(
+      (deps, [name, version]) => ({ ...deps, [name]: appuiPackages.includes(name) ? version.replaceAll("^", "") : version }),
+      {},
+    );
+    pkg.peerDependencies = fixedPeers;
+  }
```

Comment:
@calebmshafer @aruniverse, is there a reason for the caret in all `itwinjs-core` peer dependency specifications? Seems like a bug to me.

Reply:
Well, let's say `core-backend@4.0` requires a peer `core-common@^4.0`. That means it's also fine using `core-common@^4.1`. That would be fine as long as all packages respect semver, however, the packages in `itwinjs-core` don't exactly do that when speaking about `@internal` APIs - we're allowed to remove such APIs anytime we want, so `core-common@4.1` may easily have some missing API being used by `core-backend@4.0`.

Since all the packages are always released in a lockstep, I don't see why caret is needed 🤔 

> typically if the peers for 2 different pkgs dont correctly resolve to the same version its due to transitive peers not being correctly satisfied (which should be listed as dev deps)

It sounds to me like we provide bad `package.json` and expect package manager to resolve good dependency versions. This does usually happen, but is not guaranteed to...

Reply:
Resolving this comment since this is not an issue in this repository and should not stop this PR, especially since the latest changes to use separate lockfiles for each package removed the need of this hack.